opentracing-impl/src/main/java/io/opentracing/impl/AbstractActiveSpanSource.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.impl;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.ActiveSpanSource;
+import io.opentracing.Span;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * {@link AbstractActiveSpanSource} implements {@link #adopt} for {@link AbstractActiveSpan}.
+ */
+public abstract class AbstractActiveSpanSource implements ActiveSpanSource {
+
+    @Override
+    public final ActiveSpan adopt(Span span) {
+        return makeContinuation(span, new AtomicInteger(1)).activate();
+    }
+
+    protected abstract AbstractActiveSpan.Continuation makeContinuation(Span wrapped, AtomicInteger refCount);
+}
```

Comment:
Yeah, this was really just because I wanted something like this class to extend in the MDC demo thing (which will never be in master). I am happy to remove as much of the -impl stuff as possible from this PR prior to merging.